src/nvim/memline_defs.h

```diff
@@ -26,9 +26,17 @@ typedef struct ml_chunksize {
 #define ML_CHNK_DELLINE 2
 #define ML_CHNK_UPDLINE 3
 
-/*
- * the memline structure holds all the information about a memline
- */
+/// memline structure: the contents of a buffer.
+/// Essentially a tree with a branch factor of 128.
+/// Lines are stored at leaf nodes.
+/// Nodes are stored on ml_mfp (memfile_T):
+///   pointer_block: internal nodes
+///   data_block: leaf nodes
+/// Memline also has "chunks" of 800 lines that are separate from the 128-tree structure ?
```

Comment:
Primarily used to speed up `line2byte()` and `byte2line()` if I understood it correctly.